File: ServerLib/LogUpload/src/main/java/com/yc/logupload/inter/BaseUploadLog.java
package com.yc.logupload.inter;

import android.content.Context;

import java.io.File;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;

/**
 * <pre>
 *     @author yangchong
 *     email  : yangchong211@163.com
 *     time   : 2018/4/17
 *     desc   : 抽象出日志上传父类
 *     revise :
 * </pre>
 */
public abstract class BaseUploadLog implements IUploadLog{

    public Context mContext;
    public final static SimpleDateFormat mTime = new SimpleDateFormat(
            "yyyy-MM-dd HH:mm:ss:SS", Locale.getDefault());

    public BaseUploadLog(Context context) {
        mContext = context;
    }

    /**
     * 发送报告
     * @param file              file文件
     * @param onUploadListener      监听listener
     */
    protected abstract void sendReport(File file, OnUploadListener onUploadListener);

    @Override
    public void uploadFile(final File file, final String content, final OnUploadListener onUploadListener) {
        buildTitle(mContext);
        buildBody(content);
        sendReport(file, onUploadListener);
    }

    @Override
    public void uploadZipFile(File file, String content, OnUploadListener onUploadListener) {
        buildTitle(mContext);
        buildBody(content);
        sendReport(file, onUploadListener);
    }

    /**
     * 构建标题
     *
     * @param context 上下文
     * @return 返回标题的string文本
     */
    public String buildTitle(Context context) {
        return " " + context.getString(context.getApplicationInfo().labelRes) + " " +
                mTime.format(Calendar.getInstance().getTime());
    }

    /**
     * 构建正文
     * @param content           字符串
     * @return                  拼接后的内容
     */
    public String buildBody(String content) {
        return content;
    }


}

File: ServerLib/LogUpload/src/main/java/com/yc/logupload/inter/BaseUploadLog.kt
package com.yc.logupload.inter

import android.content.Context
import com.yc.logupload.config.UploadInitHelper
import java.io.File
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author yangchong
 * email  : yangchong211@163.com
 * time   : 2022/5/11
 * desc   : 抽象出日志上传父类
 * revise :
 */
abstract class BaseUploadLog() : IUploadLog {

    /**
     * 发送报告
     * @param file              file文件
     * @param onUploadListener      监听listener
     */
    protected abstract fun sendReport(file: File?, onUploadListener: OnUploadListener?)

    override fun uploadFile(file: File?, content: String?, onUploadListener: OnUploadListener?) {
        if (file == null || !file.exists()) {
            if (onUploadListener != null) {
                return onUploadListener.onError("文件不存在")
            }
        }
        UploadInitHelper.mContext?.let {
            buildTitle(it)
        }
        content?.let {
            buildBody(it)
        }
        sendReport(file, onUploadListener)
    }

    /**
     * 构建标题
     *
     * @param context 上下文
     * @return 返回标题的string文本
     */
    fun buildTitle(context: Context): String {
        return " " + context.getString(context.applicationInfo.labelRes) + " " +
                mTime.format(Calendar.getInstance().time)
    }

    /**
     * 构建正文
     * @param content           字符串
     * @return                  拼接后的内容
     */
    private fun buildBody(content: String): String {
        //获取包命
        val packageName = UploadInitHelper.mContext?.packageName
        return packageName + content
    }

    companion object {
        val mTime = SimpleDateFormat(
            "yyyy-MM-dd HH:mm:ss:SS", Locale.getDefault()
        )
    }
}